tests/pipelines/test_sklearn_classifier.py

```diff
@@ -0,0 +1,83 @@
+# flake8: noqa
+from functools import partial
+import os
+
+from pytest import mark
+from torch.optim import Adam
+from torch.utils.data import DataLoader
+
+from catalyst import data, dl
+from catalyst.contrib import datasets, models, nn
+from catalyst.data.transforms import Compose, Normalize, ToTensor
+from catalyst.settings import SETTINGS, Settings
+
+if SETTINGS.ml_required:
+    from sklearn.linear_model import LogisticRegression
+    from sklearn.metrics import log_loss
+
+
+def train_experiment():
+    # 1. train and valid loaders
+    transforms = Compose([ToTensor(), Normalize((0.1307,), (0.3081,))])
+
+    train_dataset = datasets.MnistMLDataset(root=os.getcwd(), download=True, transform=transforms)
+    sampler = data.BalanceBatchSampler(labels=train_dataset.get_labels(), p=5, k=10)
+    train_loader = DataLoader(
+        dataset=train_dataset, sampler=sampler, batch_size=sampler.batch_size
+    )
+
+    valid_dataset = datasets.MNIST(root=os.getcwd(), transform=transforms, train=False)
+    valid_loader = DataLoader(dataset=valid_dataset, batch_size=1024)
+
+    # 2. model and optimizer
+    model = models.MnistSimpleNet(out_features=16)
+    optimizer = Adam(model.parameters(), lr=0.001)
+
+    # 3. criterion with triplets sampling
+    sampler_inbatch = data.HardTripletsSampler(norm_required=False)
+    criterion = nn.TripletMarginLossWithSampler(margin=0.5, sampler_inbatch=sampler_inbatch)
+
+    # 4. training with catalyst Runner
+    class CustomRunner(dl.SupervisedRunner):
+        def handle_batch(self, batch) -> None:
+            images, targets = batch["features"].float(), batch["targets"].long()
+            features = self.model(images)
+            self.batch = {
+                "embeddings": features,
+                "targets": targets,
+            }
+
+    callbacks = [
+        dl.ControlFlowCallback(
+            dl.CriterionCallback(input_key="embeddings", target_key="targets", metric_key="loss"),
+            loaders="train",
+        ),
+        dl.SklearnClassifierCallback(
+            feature_key="embeddings",
+            target_key="targets",
+            train_loader="train",
+            valid_loader="valid",
+            sklearn_classifier_fn=LogisticRegression,
+            sklearn_metric_fn=log_loss,
+        ),
+    ]
+
+    runner = CustomRunner(input_key="features", output_key="embeddings")
+    runner.train(
+        model=model,
+        criterion=criterion,
+        optimizer=optimizer,
+        callbacks=callbacks,
+        loaders={"train": train_loader, "valid": valid_loader},
+        verbose=False,
+        valid_loader="valid",
+        valid_metric="sklearn_classifier_metric",
+        minimize_valid_metric=False,
+        num_epochs=2,
+        overfit=True,
+    )
```

Comment:
could you please add any asserts to check, that callbacks have worked out?

Reply:
[Done](https://github.com/catalyst-team/catalyst/pull/1261/commits/c7a8855c2bbcfaa829c4fd505e7847ad038fdf06)